internal/service/secretsmanager/secrets_data_source_test.go

```diff
@@ -0,0 +1,73 @@
+package secretsmanager_test
+
+import (
+	"fmt"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+)
+
+func TestAccSecretsManagerSecretsDataSource_filter(t *testing.T) {
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_secretsmanager_secret.test"
+	dataSourceName := "data.aws_secretsmanager_secrets.test"
+
+	propagationSleep := func() resource.TestCheckFunc {
+		return func(s *terraform.State) error {
+			log.Print("[DEBUG] Test: Sleep to allow secrets become visible in the list.")
+			time.Sleep(30 * time.Second)
+			return nil
+		}
+	}
```

Comment:
It takes some time for the secrets to become visible in `ListSecrets` API response.

Reply:
Secrets Manager is [eventually consistent](https://docs.aws.amazon.com/secretsmanager/latest/userguide/troubleshoot.html#troubleshoot_general_eventual-consistency) - e.g. https://github.com/hashicorp/terraform-provider-aws/issues/18406.
I'll see if there's anything I can do in the `aws_secretsmanager_secret` resource to avoid having to explicitly sleep.

Reply:
Not really 😄